lib/firmata.js

```diff
@@ -545,6 +545,17 @@ var Board = function(port, options, callback) {
         this.setSamplingInterval(settings.samplingInterval);
       }
       if (settings.skipCapabilities) {
```

Comment:
@soundanalogous it took me a bit to understand the need here, but @monteslu was kind enough to patiently expain it to me: really low power devices have trouble keeping pace with the first few messages and to reduce air traffic, we introduced the `skipCapabilties` option for those programs that want to avoid that step. Initially I thought this was pointless and broken, but again @monteslu patiently explained his use cases and I eventually came around. The issue that we initially overlooked, and that this seeks to correct, is that the `this.pins` array ends up being empty and any code that wants to interact with that will blow up, because "`this.pins[whatever].value...` cannot read property value of undefined" and so forth. So at RobotsConf we planned to fix that, but somehow it never happened, probably my own fault. @monteslu reminded me last week at JSConf and I cringed at a `getPin` and `setPin` pair, so it looks like @monteslu figured out a way to avoid icky API additions like that. Allowing user code to provide a predefined pins array, or number of "dummy" pins to create is a reasonable compromise.